### PR TITLE
feat(putPlayersInTeams): Assign teams evenly

### PR DESCRIPTION
**Pull request by Baboo7**
## Proposed changes

ticket: https://ogre-d4g.atlassian.net/browse/D4G-216

---

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

---

What types of changes does your code introduce to ogre ?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe):

## Checklist

---

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

offseason_ogre/blob/master/CONTRIBUTING.md) doc

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation and modify the README.md (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] If the feature changes current behavior, reasons why your solution is better.

## What is the current behavior?

---

-
-
-

## What is the new behavior?

---

-
-
-

## Further comments

---

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

## Where has this been tested?

---

Docker version: …
Operating System: …
node version: …
postgresql version: …

image

### CODE REVIEW

File: packages/server/src/modules/games/services/putPlayersInTeams.ts [hunk 1]
```diff
@@ -1,21 +1,61 @@
...
+  const teamCount = teamIds.length;
+
+  const basePlayerCountPerTeam = Math.floor(playerCount / teamCount);
+  const remainingPlayerCount = playerCount % teamCount;
+  const playerIdxToTeamId = flatten(
+    Array(teamCount)
+      .fill(basePlayerCountPerTeam)
+      .map(dispatchRemainingPlayerCount(remainingPlayerCount))
+      .map(createArraysOfTeamIds(teamIds))
+  );
```
Comment (louis-sanna-dev) [inline packages/server/src/modules/games/services/putPlayersInTeams.ts]: @Baboo7 [detail] peut-être qu'un flatMap pourrait nous éviter le `flatten` lodash?

(je suis pas sûr, honnetement j'ai du mal à comprendre ce code)

Comment (Baboo7) [reply packages/server/src/modules/games/services/putPlayersInTeams.ts]: Je viens de regarder `flatMap` et ça marcherait 👍 

Pour le code, j'ai essayé de le rendre le plus clair possible. L'idée c'est d'obtenir dans `playerIdxToTeamId` une liste de `teamId` qu'on peut assigner aux joueurs en utilisant le index dans le tableau `players`. De cette manière on peut facilement assigner une équipe à un joueur comme c'est fait ligne 46. 

Pour 3 équipes d'ids `1`, `2` et `3` et 8 joueurs, `playerIdxToTeamId` contient 8 entrées : `[1, 1, 1, 2, 2, 2, 3, 3]`. (les joueurs d'index 0 à 2 seront dans l'équipe 1, ceux d'index 3 à 5 dans la 2...)

Comment (louis-sanna-dev) [reply packages/server/src/modules/games/services/putPlayersInTeams.ts]: @Baboo7 pk on peut pas faire un truc du genre:

```ts
const teamPlayerAssignment = players.map((player, index) => ({
    player,
    teamId: teamIds[index % teamIds.length],
  }));
```

**Review by louis-sanna-dev — APPROVED** [review]

File: packages/server/src/modules/games/services/putPlayersInTeams.ts (hunk 1 above)
Comment (Baboo7) [reply packages/server/src/modules/games/services/putPlayersInTeams.ts]: Smart !

**Pull request merged**